go.mod

```diff
@@ -12,23 +12,23 @@ require (
 	github.com/gophercloud/utils v0.0.0-20221207145018-e8fba78967ca
 	github.com/hashicorp/go-version v1.4.0
 	github.com/onsi/ginkgo/v2 v2.11.0
-	github.com/onsi/gomega v1.27.8
+	github.com/onsi/gomega v1.27.10
 	github.com/prometheus/client_golang v1.16.0
 	github.com/spf13/pflag v1.0.5
 	golang.org/x/crypto v0.11.0
 	golang.org/x/text v0.11.0
 	gopkg.in/ini.v1 v1.67.0
-	k8s.io/api v0.27.2
-	k8s.io/apiextensions-apiserver v0.27.2
-	k8s.io/apimachinery v0.27.2
-	k8s.io/client-go v0.27.2
-	k8s.io/component-base v0.27.2
-	k8s.io/klog/v2 v2.90.1
+	k8s.io/api v0.28.2
+	k8s.io/apiextensions-apiserver v0.28.0
+	k8s.io/apimachinery v0.28.2
+	k8s.io/client-go v0.28.2
+	k8s.io/component-base v0.28.1
+	k8s.io/klog/v2 v2.100.1
 	k8s.io/kubernetes v1.27.2
-	k8s.io/utils v0.0.0-20230313181309-38a27ef9d749
-	sigs.k8s.io/cluster-api v1.5.1
-	sigs.k8s.io/cluster-api/test v1.5.1
-	sigs.k8s.io/controller-runtime v0.15.1
+	k8s.io/utils v0.0.0-20230406110748-d93618cff8a2
+	sigs.k8s.io/cluster-api v1.5.2
+	sigs.k8s.io/cluster-api/test v1.5.2
+	sigs.k8s.io/controller-runtime v0.16.2
```

Comment:
Please make sure you bump controller-runtime to `v0.16.3`+ to get the RapidReset mitigation. This is what CAPI v1.6 will very likely get anyway: https://github.com/kubernetes-sigs/cluster-api/blob/dd553a2e82b4d76f6f9bf1f0f66b37fd1d640416/go.mod#L49

Reply:
ack, I'll keep that in mind when we come back to this but first we need a new CAPI release.

Reply:
I'll need to update this PR once CAPI V1.6.0, end of November.
See https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/1734